sway/ipc-json.c

```diff
@@ -1003,6 +1004,12 @@ json_object *ipc_json_describe_input(struct sway_input_device *device) {
 	}
 
 	if (wlr_input_device_is_libinput(device->wlr_device)) {
+		struct input_config *ic = input_device_get_config(device);
```

Comment:
The scroll factor is also applied when the device isn't coming from libinput (e.g. virtual pointers for VNC, running with wlroots' Wayland or X11 backend, etc). I think it would make sense to have a new `if (device->wlr_device->type == WLR_INPUT_DEVICE_POINTER)` block.